rpy2-robjects/src/rpy2/ipython/rmagic.py

```diff
@@ -57,84 +62,449 @@
+        pass
+
+    def new_device(
+            self,
+            width=None,
+            height=None,
+            pointsize=None,
+            bg=None,
+            units=None,
+            res=None
+    ):
+
+        if width:
+            width = self._options_new_device.width
+        if height:
+            height = self._options_new_device.height
+        if pointsize:
+            pointsize = self._options_new_device.pointsize
+        if bg:
+            bg = self._options_new_device.bg
+        if units:
+            units = self._options_new_device.units
+        if res:
+            units = self._options_new_device.res
+
+        res = Options._fix_graphics_resolution(units, res)
+        tmpd = tempfile.mkdtemp()
+        tmpd_fix_slashes = tmpd.replace('\\', '/')
+
+        # TODO: create a function for x if x else MissingArg.
+        self.get_rfunc()(
+            f'{tmpd_fix_slashes}/Rplots%03d.{self.extension}',
+            width=width if width else ri.MissingArg,
+            height=height if height else ri.MissingArg,
+            pointsize=pointsize if pointsize else ri.MissingArg,
+            bg=bg if bg else ri.MissingArg,
+            units=units if units else ri.MissingArg,
+            res=res if res else ri.MissingArg
+        )
+        return tmpd
+
+    def display_filename(self, filename):
+        img = IPython.display.Image(filename=filename)
+        IPython.display.display(img,
+                                metadata={})
+        return img
+
+    def iter_displayed_path(self, path):
+        filename_seq = sorted(
+            glob(os.path.join(path, f'Rplots*.{self.extension}'))
+        )
+
+        for imgfile in filename_seq:
+            if os.stat(imgfile).st_size >= 1000:
+                img = self.display_filename(imgfile)
+                _sync_console()
+                yield img
+
+
+class GraphicsDeviceSVG(FileDevice, GraphicsDevice):
+
+    class _OptionsNewDevice(GraphicsDevice._OptionsNewDevice):
+        width: typing.Optional[float]
+        height: typing.Optional[float]
+        pointsize: typing.Optional[float]
+        bg: typing.Optional[str]
+        units: typing.Optional[str]
+        res: typing.Optional[str]
+
+    class _OptionsDisplay(GraphicsDevice._OptionsDisplay):
+        isolate_svgs: bool = True
+
+    def new_device(
+            self,
+            width=None,
+            height=None,
+            pointsize=None,
+            bg=None,
+            units=None,
+            res=None
+    ):
+        if width:
+            width = self._options_new_device.width
+        if height:
+            height = self._options_new_device.height
+        if pointsize:
+            pointsize = self._options_new_device.pointsize
+        if bg:
+            bg = self._options_new_device.bg
+        if units:
+            units = self._options_new_device.units
+        if res:
+            units = self._options_new_device.res
```

Comment:
Seems like a typo, should be `res = ` (but maybe `units` and `res` shouldn't be in `GraphicsDeviceSVG` at all, to match previous code).